sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -3,209 +3,253 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # -------------------------------------------------------------------------
-
 import json
+from typing import overload, List, Tuple
 from azure.appconfiguration import AzureAppConfigurationClient
 from azure.keyvault.secrets import SecretClient, KeyVaultSecretIdentifier
-from azure.core.exceptions import ResourceNotFoundError
+from ._azureappconfigurationkeyvaultoptions import AzureAppConfigurationKeyVaultOptions
 from ._settingselector import SettingSelector
-from ._azure_appconfiguration_provider_error import KeyVaultReferenceError
 from ._constants import KEY_VAULT_REFERENCE_CONTENT_TYPE
 
 from ._user_agent import USER_AGENT
 
-
-class AzureAppConfigurationProvider:
+@overload
+def load_provider(endpoint: str, credential: str, **kwargs):
```

Comment:
What does the usage look like after this change?

Reply:
You can see the usage in the samples, it looks not different than now really, minus being part of the class.